dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/MeasureFilter.java

```diff
@@ -35,5 +35,43 @@
  */
 public enum MeasureFilter
 {
-    EQ, GT, GE, LT, LE
+    EQ, GT, GE, LT, LE;
+
+    /**
+     * Checks the constraint using x and y as the values and uses the current
+     * measurefilter instance for comparison.
+     * 
+     * @param x The first double value to be compared
+     * @param y The second double value to be compared
+     * @return true if the constraint/filter is satisfied when x is compared
+     *         with y.
+     */
+    public boolean checkConstraint( Double x, Double y )
```

Comment:
Perhaps call `measureIsValid` or something that illustrates the returned method value?

Reply:
Yep. Sounds good.